components/brave_news/browser/feed_v2_builder.cc

```diff
@@ -629,9 +642,13 @@ void FeedV2Builder::BuildFollowingFeed(
       subscriptions, {.signals = true},
       mojom::FeedV2Type::NewFollowing(mojom::FeedV2FollowingType::New()),
       base::BindOnce([](FeedGenerationInfo info) {
-        return GenerateBasicFeed(std::move(info),
-                                 base::BindRepeating(&PickRoulette),
-                                 base::BindRepeating(&PickRoulette));
+        auto subscriptions = info.subscriptions();
+        auto top_stories = GetTopStoryUrls(info.topics());
```

Comment:
could we make this a boolean property of the article object? So we only have to compare it once?

Reply:
We can, but I don't think it'll make a massive difference and I'd prefer the `FeedItemMetaData` object to be a fairly 1-1 mapping of whats in the JSON.

I think it'll make things more complicated, as we update the top stories and the Articles separately so the data won't always be consistent there (i.e. if we update articles but not topics or topics but not articles) so I think it makes the logic a bit less straightforward.